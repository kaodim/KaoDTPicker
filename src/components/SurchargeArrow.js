import SurchargeArrow from '../svg/ic-surcharge.svg'

export default SurchargeArrow
