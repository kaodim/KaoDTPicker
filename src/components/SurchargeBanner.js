import React from 'react'
import PropTypes from 'prop-types'
import SurchargeArrow from './SurchargeArrow'

const SurchargeBanner = ({ disableIcon, label, surchargeGif }) => {
  if (disableIcon) {
    return (
      <section className="kld-surcharge">
        <span className="kld-surcharge__label">{label}</span>
      </section>
    )
  }
  return (
    <section className="kld-surcharge">
      {surchargeGif ? (
        <img src={surchargeGif} className="kld-surcharge__icon" />
      ) : (
        <SurchargeArrow className="kld-surcharge__icon" />
      )}
      <span className="kld-surcharge__label">{label}</span>
    </section>
  )
}

SurchargeBanner.defaultProps = {
  disableIcon: false,
  label: '',
  surchargeGif: ''
}

SurchargeBanner.propTypes = {
  disableIcon: PropTypes.bool,
  label: PropTypes.string,
  surchargeGif: PropTypes.string
}

export default SurchargeBanner
