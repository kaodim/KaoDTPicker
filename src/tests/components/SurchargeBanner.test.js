import React from 'react'
import { shallow } from 'enzyme'
import SurchargeBanner from '../../components/SurchargeBanner'
import SurchargeArrow from '../../components/SurchargeArrow'

let wrapper
const bannerText = 'Hello there.'

beforeEach(() => {
  wrapper = shallow(<SurchargeBanner />)
})

test('should render SurchargeBanner with svg', () => {
  expect(wrapper.find(SurchargeArrow).length).toBe(1)
})

test('should render SurchargeBanner with gif', () => {
  wrapper.setProps({
    surchargeGif: 'whatever.gif'
  })
  expect(wrapper.find('img').length).toBe(1)
})

test('should render SurchargeBanner without svg', () => {
  wrapper.setProps({
    disableIcon: true
  })
  expect(wrapper.find(SurchargeArrow).length).toBe(0)
})

test('should render SurchargeBanner with banner text', () => {
  wrapper.setProps({
    label: bannerText
  })
  expect(wrapper).toMatchSnapshot()
})

test('should render SurchargeBanner with banner text (no svg)', () => {
  wrapper.setProps({
    disableIcon: true,
    label: bannerText
  })
  expect(wrapper).toMatchSnapshot()
})
