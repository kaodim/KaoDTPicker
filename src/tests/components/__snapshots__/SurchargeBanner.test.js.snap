// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SurchargeBanner with banner text (no svg) 1`] = `
<section
  className="kld-surcharge"
>
  <span
    className="kld-surcharge__label"
  >
    Hello there.
  </span>
</section>
`;

exports[`should render SurchargeBanner with banner text 1`] = `
<section
  className="kld-surcharge"
>
  <SurchargeArrow
    className="kld-surcharge__icon"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  />
  <span
    className="kld-surcharge__label"
  >
    Hello there.
  </span>
</section>
`;
